File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.kt
import kotlin.math.max
import kotlin.Int.Companion.MAX_VALUE

class Solution {
    fun maxProfit(prices: IntArray): Int {
        var maxP = 0
        var minP = MAX_VALUE
        for(p in prices){
            minP = min(minP,p)
            maxP = max(maxP, p-minP)
        }
       return maxP
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int Min = INT_MAX;
        int ans = 0;
        
        for (int i = 0; i < n; i++) {
            if (prices[i] < Min) {
                Min = prices[i];
            } else {
                ans = max(ans, prices[i] - Min);
            }
        }
        
        return ans;
    }
};